File: Java/Greaterfigure.java
//PROGRAM TO FIND THE GREATER FIGUE

import java.util.*;

abstract class Figure
	{
	abstract public double area();
	abstract public void show();
	}

class Rectangle extends Figure
	{
	private double l,b;
	public double area()
		{return l*b;}
	public Rectangle(double l,double b)
		{this.l=l;this.b=b;}
	public void show()
		{System.out.println("Rectangle:");
		System.out.println("Lenght="+l);
		System.out.println("Breadth="+b);}

class Square extends Figure
	{
	private double l;
	public double area()
		{return l*l;}
	public Square(double l)
		{this.l=l;}
	public void show()
		{System.out.println("Square:");
		System.out.println("Length="+l);}
	}

class Greaterfigure
	{
	public static void main(String args[])
		{
		Scanner sc=new Scanner(System.in);
		double lrect,b,lsquare;
		System.out.print("Enter the length and bredth of rectangle:");
		lrect=sc.nextDouble();
		b=sc.nextDouble();
		System.out.print("\nEnter the length of square:");
		lsquare=sc.nextDouble();
		Rectangle ret=new Rectangle(lrect,b);
		Square sq=new Square(lsquare);
		Figure fig;
		fig=greater_figure(ret,sq);
		fig.show();
		System.out.println("\nArea:"+fig.area());
		}
	static Figure greater_figure(Rectangle r,Square s)
		{
		if(r.area()>s.area())
		return r;
		return s;
		}
	}

File: C++ Programs/Object Oriented Program/Friend Function/GREATER_FIGURE.cpp
//PROGRAM TO FIND GREATER FIGURE USING FRIEND FUNCTION

#include <iostream>
using namespace std;

class Square;
class Rectangle
	{
		private:int l,b;
		
		public: Rectangle(){l=b=0;}
				Rectangle(int a,int b)
					{
					l=a;
					this->b=b;
					}
				Rectangle(Rectangle &a)
					{
					l=a.l;
					b=a.b;
					}
				~Rectangle(){}
				void input()
					{
						cout<<"\nEnter the length and bredth:";
						cin>>l>>b;
					}
				void output()
					{
						cout<<"\nL="<<l
							<<"\nB="<<b;
					}
				friend void greater(Rectangle &,Square &);
			};
			
class Square
	{
		private: int l;
		
		public: Square(){l=0;}
				Square(int a){l=a;}
				Square(Square &a){l=a.l;}
				~Square(){}
				void input()
					{
						cout<<"\nEnter the length of square:";
						cin>>l;
					}
				void output()
					{
						cout<<"\nL="<<l;
					}
				friend void greater(Rectangle &,Square &);
		};
		
void greater(Rectangle &r,Square &s)
	{
		if(r.l*r.b>s.l*s.l)
			cout<<"\nArea of rectangle:"<<r.l*r.b;
		else
			cout<<"\nArea of square:"<<s.l*s.l;
	}
	
int main()
	{
		int x,y;
		cout<<"\nEnter the length & bredth of recangle:";
		cin>>x>>y;
		Rectangle rect(x,y);
		cout<<"\nEnter the length of square:";
		cin>>x;
		Square sq(x);
		cout<<"\nGreater Figure:";
		::greater(rect,sq);
		return 0;
	}
